joblib/parallel.py

```diff
@@ -249,6 +252,19 @@ def __call__(self):
             return [func(*args, **kwargs)
                     for func, args, kwargs in self.items]
 
+    def __reduce__(self):
+        if self._reducer_extra is not None:
+            # reducer_extra should be "not None" only with loky backends.
+            # Relevant implementation detail: the following lines are
+            # thread-safe, meaning that the context of the temporary folder
+            # manager will not be changed in between now and the end of the
+            # BatchedCall pickling. The reason is that pickling (the only place
+            # where set_context is used) is done from a single thread (the
+            # executor manager thread).
```

Comment:
This comment is great, it took me a while to realise this and I should have read it before !
To be complete, I think this is only done by the `queue_feeder_thread` :)